pkg/loop/internal/relayer/pluginprovider/chainreader/chain_reader_test.go

```diff
@@ -383,6 +408,10 @@ func (f *fakeChainReader) SetBatchLatestValues(batchCallEntry BatchCallEntry) {
 }
 
 func (f *fakeChainReader) GetLatestValue(_ context.Context, contractName, method string, confidenceLevel primitives.ConfidenceLevel, params, returnVal any) error {
+	if !f.isStarted {
```

Comment:
These reads of `f.isStarted` need to be synchronized with the writes in `Start`/`Close`. How about using an atomic variable instead?

Reply:
https://pkg.go.dev/sync/atomic#Bool

Reply:
Thanks! Didn't know I could do this 😄